README.md

```diff
@@ -1,55 +1,142 @@
 
 The app uses [Amazon Key Management Service (KMS)](https://aws.amazon.com/kms/) to create and control cryptographic keys and the [AWS SDK for Python (boto3)](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms.html) to call KMS.
+During development and testing, you can also use [LocalStack](https://www.localstack.cloud/) to run a localized environment that simulates interactions with AWS.
 
 ### About CSRs
 
-In addition to being an example of using the Python library, this app shows how to generate a certificate signing request (CSR), a message sent to a certificate authority (CA) to request the signing of a public key and associated information. 
+In addition to being an example of using the Python library, this app shows how to generate a certificate signing request (CSR), a message sent to a certificate authority (CA) to request the signing of a public key and associated information.
 
-A CSR comprises a public key, as well as a common name, organization, city, state, country, and e-mail address. Not all of these fields are required and will vary depending with the assurance level of your certificate. 
+A CSR comprises a public key, as well as a common name, organization, city, state, country, and e-mail address. Not all of these fields are required and will vary depending with the assurance level of your certificate.
 
-You sign the CSR with your private key; this proves to the CA that you have control of the private key that corresponds to the public key included in the CSR. Once the requested information in a CSR passes a vetting process and domain control is established, the CA may sign the public key to indicate that it can be publicly trusted. 
+You sign the CSR with your private key; this proves to the CA that you have control of the private key that corresponds to the public key included in the CSR. Once the requested information in a CSR passes a vetting process and domain control is established, the CA may sign the public key to indicate that it can be publicly trusted.
 
-## Prerequisites
+## Setup to use Docker
+
+The example code from this repository can run in Docker containers with default  configurations. The Docker containers use LocalStack to run a localized environment that simulates interactions with AWS. This is the quickest way to spin up a working development environment (without doing additional configuration.) 
+
+NOTE: This is a development setup and should not be deployed as-is to a production environment.
+
+### Supported Docker versions
+
+The example runs with Docker Desktop version 4.34.3 (170107) or later.
+
+### Run the local setup
```

Comment:
This might be confusing since the later section is called "Run the example locally". Suggest titling this section "Run the app in Docker container" (since it covers more than just setup) or something like that.